go-controller/pkg/metrics/metrics.go

```diff
@@ -126,33 +132,53 @@ func coverageShowMetricsUpdater(component string) {
 		}
 		coverageShowMetricsMap := componentCoverageShowMetricsMap[component]
 		for metricName, metricInfo := range coverageShowMetricsMap {
-			if metricName == "packet_in" {
-				metricName = "flow_extract"
+			var metricValue float64
+			if metricInfo.srcName != "" {
+				metricName = metricInfo.srcName
 			}
-			if value, ok := coverageShowOutputMap[metricName]; ok {
-				count := parseMetricToFloat(component, metricName, value)
-				metricInfo.metric.Set(count)
+			if metricInfo.aggregateFrom != nil {
+				for _, aggregateMetricName := range metricInfo.aggregateFrom {
+					if value, ok := coverageShowOutputMap[aggregateMetricName]; ok {
+						metricValue += parseMetricToFloat(component, aggregateMetricName, value)
+					}
+				}
 			} else {
-				metricInfo.metric.Set(0)
+				if value, ok := coverageShowOutputMap[metricName]; ok {
+					metricValue = parseMetricToFloat(component, metricName, value)
+				}
 			}
+			metricInfo.metric.Set(metricValue)
 		}
 	}
 }
 
-// StartMetricsServer runs the prometheus listner so that metrics can be collected
+func checkPodRunsOnGivenNode(clientset *kubernetes.Clientset, label, k8sNodeName string) (bool, error) {
+	pods, err := clientset.CoreV1().Pods(config.Kubernetes.OVNConfigNamespace).List(context.TODO(), metav1.ListOptions{
```

Comment:
we can implement the wait.Pollinmediate loop here, so we can use the function directly.
In case of waiting for multiple pods we can also spawn to go routines
    var waitgroup sync.WaitGroup
    waitgroup.Add(1)
    go func() {
        podExist, err := checkPodRunsOnGivenNode( pod1)
        waitgroup.Done()
    }()
    waitgroup.Add(1)
    go func() {
        podExist, err := checkPodRunsOnGivenNode( pod2)
        waitgroup.Done()
    }()     
    waitgroup.Wait()
```

Reply:
Thanks @aojea. I have fixed the logic now that we only need to check for one label, so we will not need the WaitGroup for now.